bindings/java/devdoc/java_out_of_process_hld.md

```diff
@@ -0,0 +1,122 @@
+Java Out of Process Module API
+==============================
+
+Overview
+--------
+
+This document describes the API that enables communication between an out of process Java module and Azure IoT Gateway. 
+An out of process module should implement `IGatewayModule` interface or extend the `GatewayModule` abstract class as it does for an in-process module, more details in [Java binding high level design](java_binding_hld.md). In addition, the API provides `GatewayModuleProxy` which is a proxy for the IGatewayModule implementation and handles the communication between the Gateway and the out of process module. 
+`GatewayModuleProxy` contains the methods to *attach* a module to the Gateway, *listen* for messages and *detach* from the Gateway. 
+
+There are two types of messages that are exchanged between the Gateway and the out of process module:
+
+-  *control* messages that are used to signal `CREATE`, `START` and `DESTROY` of a module. These messages are always send from the Gateway to the module.
+-  *data* messages which are send between modules via the Gateway.
+
+
+![](java_oop_module_hld.png)
```

Comment:
I updated the diagram to use "control message" and point arrow down. I also changed the upper part is "nanomsg sockets" instead of Azure IoT Gateway and the Gateway is on the left side.